### ISSUE TITLE
Unknown compiler option 'compileOnSave'

### ISSUE THREAD

**Issue by mani-mishra** [emberwatch/ember-cli-typescript#23]
Running ```tsc``` directly, ```compileOnSave``` works, but ```ember serve``` command throws an error. 

Also the compiler seems to be running 3 times (using default tsconfig.json as recommended by the repo)

```
Livereload server on http://localhost:49153
Serving on http://localhost:4200/
Unknown compiler option 'compileOnSave'.
Unknown compiler option 'compileOnSave'.
Unknown compiler option 'compileOnSave'.
```

```
ember --version
ember-cli: 2.12.1
node: 7.4.0
os: linux x64
```
and
```
tsc -v
Version 2.2.2
```

package.json TS stuff

```
 "@types/ember": "^2.7.37",
 "@types/underscore": "^1.8.0",
 "ember-cli-typescript": "github:emberwatch/ember-cli-typescript",
"typescript": "^2.2.2"
```

**Comment by chriskrycho** [comment]
Thanks for the writeup, @manimis902! I'll see if I can chase it down – probably Saturday before I get to it. Can you give me the output from `ember --version` and the versions of anything TypeScript in your package.json so I can be sure to reproduce?

Note that I *suspect* this is a broccoli-typescript-compiler issue, upstream of this add-on, but I'll confirm it either way this weekend.

**Comment by manimis902** [comment]
thanks @chriskrycho Even I thought of opening the issue there first :) Below are the required info.
I'll add them in description as well.

```
ember --version
ember-cli: 2.12.1
node: 7.4.0
os: linux x64
```
and
```
tsc -v
Version 2.2.2
```

**Comment by chriskrycho** [comment]
Following up on this:

1. #45 should resolve the actual bugs.
2. You *shouldn't* need the `compileOnSave` option turned on if you're using `ember serve`, but it also shouldn't do any harm: the `broccoli-typescript-compiler` add-on should treat it appropriately, I *imagine*. If after #45 is out that still throws an error, let me know, but the changes for the 2.0 of that package should handle this scenario I believe.

**Comment by chriskrycho** [comment]
This should be resolved in the 1.0 beta. Please test with `"ember-cli-typescript": "next"` and let me know if it recurs.
